### PR TITLE
demo

### PR DESCRIPTION
**Pull request by jacobderosa**
added demo script, instructions, and example config files

### CODE REVIEW

File: demo/input_Jacksonville.ini [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+[params]
+gsd = 0.25
+
+[material]
+model_fpath = /models/Rutgers\ Material\ Segmentation\ Model/model_D4_20.pth.tar
+cuda = True
+
+[roof]
+model_dir = /models/Columbia Geon Segmentation Model
```
Comment (BryonLewis) [inline demo/input_Jacksonville.ini:18]: I think in testing we may want to add back in the backspace for this dir and other configs as well.

**Comment by jacobderosa** [comment]
@borovik135

File: demo/danesfield_demo.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+def run(args):
+	parser = argparse.ArgumentParser(description="Download WV3 images to run the Danesfield pipeline")
+	parser.add_argument('--site', help='CORE3D site to download images for', \
+						default='Jacksonville')
+	parser.add_argument('--img_path', help='Location where WV3 images should be downloaded to', \
+						default=f'{os.getcwd()}/imgpath')
+	parser.add_argument('--run', help='Run Danesfield after downloading images', \
+						action='store_true')
+	parser.add_argument('--out_path', help='Location that Danesfield should use as an output directory', default='outdir')
```
Comment (borovik135) [inline demo/danesfield_demo.py:16]: Watch the line lengths, as code formatting guidelines suggest max 80 characters per line.

File: demo/danesfield_demo.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	parser.add_argument('--config_dir', help='Location where Danesfield configuration files can be found. \
+											You must use input_<region>.ini and <region>_config.json for \
+											Danesfield and Vissat config names, respectively.', default='configdir')
+	parser.add_argument('--show_sites', help='List available CORE3D sites and exit', \
+						action='store_true')
+	parser.add_argument('--model_path', help='Location of necessary model files for Danesfield', default="Models")
```
Comment (borovik135) [inline demo/danesfield_demo.py:22]: Consider communicating default values to the user, e.g. in help='a short description of the parameter; default: %(default)s'

File: demo/danesfield_demo.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+		subprocess.run(cmd_args)
+		extract_command = ['unzip', f'models.zip']
+		subprocess.run(extract_command)
+		print('Models Pulled and Extracted')
+
+	# Create necessary directories
+	os.makedirs(args.out_path, exist_ok=True)
+	os.makedirs(args.img_path, exist_ok=True)	
+	os.makedirs(args.config_dir, exist_ok=True)
+	#os.makedirs(args.model_path, exist_ok=True)
```
Comment (borovik135) [inline demo/danesfield_demo.py:55]: Do we still need this and perhaps other commented-out code?

File: demo/danesfield_demo.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+		config = configparser.ConfigParser()
+		config.read(os.path.join(args.config_dir, 'input_'+args.site+'.ini'))
+		config['paths']['p3d_fpath'] = os.path.join('/workdir', args.site, 'cloud.las')
+		config['paths']['work_dir'] = os.path.join('/workdir', args.site)
+		config['paths']['imagery_dir'] = os.path.join('/mnt', args.site, 'WV3')
+		config['paths']['aoi_config'] = os.path.join('/configs', args.site+'_config.json')
+		config['aoi']['name'] = args.site.lower()
+		config['roof']['model_dir'] = 'models/Columbia Geon Segmentation Model'
+		config['roof']['model_prefix'] = 'dayton_geon'
+		config['material']['model_fpath'] = '/models/Rutgers\\ Material\\ Segmentation\\ Model/model_D4_20.pth.tar'
```
Comment (borovik135) [inline demo/danesfield_demo.py:102]: Consider using double-quotes instead of escaping spaces.

File: demo/demo_instructions.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	python danesfield/tools/run_danesfield.py --image --vissat \
+	/configs/input_Jacksonville.ini
+
+This can be used as a starting point for manually running the Danesfield Docker container in other ways. Notably, the ``--image`` option indicates that the pipeline should start with satellite images (instead of a point cloud) and the ``--vissat`` option indicates that VisSat should be used to generate a point cloud from the images. To process a point cloud, leave off both of these options and in the Danesfield configuration (see section below) set ``p3d_path`` to the point cloud file to use. 
+
+Configuring Danesfield
+----------------------
+
+Two configuration files are needed for Danesfield. The first is required to be named ``input_<region>.ini``, where <region> is replaced by the name of teh site. An example of what information it needs can be found at https://github.com/Kitware/Danesfield/blob/master/tools/example_vissat_config.json.
+The only fields that need to be filled out in this file are ``gsd`` and ``cuda``. The first is the desired ground sample distance (default .25), and the second is whether or not Cuda is available.
```
Comment (borovik135) [inline demo/demo_instructions.rst:48]: Consider all capitals in CUDA.

File: demo/input_Jacksonville.ini [hunk 7]
```diff
@@ -0,0 +1,20 @@
+[paths]
+p3d_fpath = /workdir/Jacksonville/cloud.las
+work_dir = /workdir/Jacksonville
+imagery_dir = /mnt/Jacksonville/WV3
+aoi_config = /configs/Jacksonville_config.json
```
Comment (borovik135) [inline demo/input_Jacksonville.ini:5]: Consider referencing container mount points.

File: demo/input_Jacksonville.ini [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+[aoi]
+name = jacksonville
+
+[params]
+gsd = 0.25
+
+[material]
+model_fpath = /models/Rutgers\ Material\ Segmentation\ Model/model_D4_20.pth.tar
```
Comment (borovik135) [inline demo/input_Jacksonville.ini:14]: Consider enclosing in double quotes instead of escaping spaces.

File: demo/input_Jacksonville.ini (hunk 1 above)
Comment (borovik135) [inline demo/input_Jacksonville.ini:18]: Enclose in double quotes paths that contain spaces.

File: demo/Jacksonville_config.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
+{
+  "dataset_dir": "/mnt/Jacksonville/WV3/PAN",
```
[the remark this answers is not in the document]
Comment (jacobderosa) [reply demo/Jacksonville_config.json:2]: Addressed in updated instructions

**Review by borovik135 — APPROVED** [review]
Most re commendations and comments appear to have been addressed, we can land it now, I suppose.

**Review by mleotta — APPROVED** [review]

**Pull request merged**